packages/core/src/shipping/strategies/stripe-upe/stripe-upe-shipping-strategy.ts

```diff
@@ -92,8 +92,32 @@ export default class StripeUPEShippingStrategy implements ShippingStrategy {
                 appearance,
             });
 
+        const shipping = this._store.getState().shippingAddress.getShippingAddress();
+        const stripeState = shipping?.stateOrProvinceCode && shipping?.countryCode?
+            getStripeState(shipping?.countryCode, shipping?.stateOrProvinceCode) : shipping?.stateOrProvinceCode;
```

Comment:
@animesh1987, yes because It's necessary to know the country code here to avoid calling getStripeState(It's a mapper made on the JS repository) and send the default value of  stateOrProvinceCode in case it's non-existing